provisioning/device/samples/ProvisioningDeviceClientX509/Program.cs

```diff
@@ -70,10 +70,36 @@ public static void Main(string[] args)
             s_idScope = args[0];
 
             string certificatePassword = ReadCertificatePassword();
-            using (var certificate = new X509Certificate2(s_certificateFileName, certificatePassword))
+
+            var certificateCollection = new X509Certificate2Collection();
+            certificateCollection.Import(s_certificateFileName, certificatePassword, X509KeyStorageFlags.UserKeySet);
+
+            X509Certificate2 certificate = null;
+
+            foreach (X509Certificate2 element in certificateCollection)
             {
-                RunSample(certificate).GetAwaiter().GetResult();
+                Console.WriteLine($"Found certificate: {element?.Thumbprint} {element?.Subject}; PrivateKey: {element?.HasPrivateKey}");
+                if (certificate == null && element.HasPrivateKey)
+                {
+                    certificate = element;
```

Comment:
Consider a "break;" here, since once you get one match there's no need continuing to enumerate.

Reply:
@jspaith The reason the code isn't breaking is to ensure all other X509 certs are disposed in the loop.